File: 643-maximum-average-subarray-i/maximum-average-subarray-i.cpp
class Solution {
public:
    double findMaxAverage(vector<int>& nums, int k) {
        double winsum=0;
        for(int i=0;i<k;i++){
            winsum+=nums[i];
        }double max_sum=winsum;
        for(int i=k;i<nums.size();i++){
            winsum+=nums[i]-nums[i-k];
            max_sum=max(max_sum,winsum);
        }
        return max_sum/k;
    }
};

File: 643-maximum-average-subarray-i/maximum-average-subarray-i.java
class Solution {
    public double findMaxAverage(int[] nums, int k) {
        int n = nums.length;
        double max = -1000000;
        for(int start = 0;start<=n-k;start++){
            int sum = 0;
            for(int end = start;end<start+k;end++){
                sum = sum+nums[end];
            }
            double avg = (double) sum/k;
            if(avg>max){
                max = avg;
            }
        }
        return max;
    }
}